File: sol/2657/2657.cpp
struct Solution {
    vector<int> findThePrefixCommonArray(vector<int>& A, vector<int>& B) {
        int n = A.size(), count = 0;
        vector<int> res(n, 0);
        vector<short> freq(n + 1, 0);
        for (int i = 0; i < n; ++i){
            ++freq[A[i]];
            count += (freq[A[i]] == 2);
            ++freq[B[i]];
            count += (freq[B[i]] == 2);
            res[i] = count;
        }
        return res;
    }
};

File: sol/2657/2657.py
class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:
        n = len(A)
        aSeen = [False] * (n + 1)
        bSeen = [False] * (n + 1)
        res = [0] * n
        count = 0
        for i in range(n):
            if A[i] == B[i]:
                count += 1
            else:
                count += aSeen[B[i]]
                count += bSeen[A[i]]
            aSeen[A[i]] = True
            bSeen[B[i]] = True
            res[i] = count
        return res
